nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -1457,17 +1489,29 @@ impl DataStore {
         // Fetch records of cabooses found.
         let inv_caboose_rows = {
             use db::schema::inv_caboose::dsl;
-            dsl::inv_caboose
+
+            let mut cabooses = Vec::new();
+
+            let mut paginator = Paginator::new(batch_size);
+            while let Some(p) = paginator.next() {
+                let mut batch = paginated(
+                    dsl::inv_caboose,
+                    dsl::time_collected,
+                    &p.current_pagparams(),
+                )
                 .filter(dsl::inv_collection_id.eq(id))
-                .limit(sql_limit)
                 .select(InvCaboose::as_select())
                 .load_async(&*conn)
                 .await
                 .map_err(|e| {
                     public_error_from_diesel(e, ErrorHandler::Server)
-                })?
+                })?;
+                paginator = p.found_batch(&batch, &|row| row.time_collected);
```

Comment:
Same -- but this is trickier because I think the unique key is really (`inv_collection_id` (which is implied by the filter), `hw_baseboard_id`, `which`).  Maybe we can use `paginated_multicolumn` here?